File: sample/hello.kt
class Greeter(val name: String) {
    fun greet() {
        println("Hello, $name from Kotlin!")
    }
}

fun calculateSum(numbers: List<Int>): Int {
    return numbers.sum()
}

fun main() {
    val greeter = Greeter("World")
    greeter.greet()

    val numbers = listOf(1, 2, 3, 4, 5)

    // Higher-order functions
    val squares = numbers.map { it * it }
    val evenNumbers = numbers.filter { it % 2 == 0 }

    println("Sum: ${calculateSum(numbers)}")
    println("Squares: ${squares.joinToString(", ")}")
    println("Even numbers: ${evenNumbers.joinToString(", ")}")

    // When expression (pattern matching)
    numbers.forEach { n ->
        when {
            n % 2 == 0 -> println("$n is even")
            else -> println("$n is odd")
        }
    }

    // Data class example
    data class Person(val name: String, val age: Int)
    val person = Person("Alice", 30)
    println("Person: $person")

    // Nullable types
    var nullable: String? = "Hello"
    println("Length: ${nullable?.length}")
}

File: sample/hello.hs
module Main where

-- Data type definition
data Greeter = Greeter { name :: String }

-- Function to greet
greet :: Greeter -> IO ()
greet (Greeter name) = putStrLn $ "Hello, " ++ name ++ " from Haskell!"

-- Calculate sum using fold
calculateSum :: [Int] -> Int
calculateSum = foldr (+) 0

-- Calculate squares using map
squares :: [Int] -> [Int]
squares = map (^2)

-- Filter even numbers
evenNumbers :: [Int] -> [Int]
evenNumbers = filter even

main :: IO ()
main = do
    let greeter = Greeter "World"
    greet greeter

    let numbers = [1, 2, 3, 4, 5]
    putStrLn $ "Sum: " ++ show (calculateSum numbers)
    putStrLn $ "Squares: " ++ show (squares numbers)
    putStrLn $ "Even numbers: " ++ show (evenNumbers numbers)

    -- List comprehension
    let squaresList = [x^2 | x <- numbers]
    putStrLn $ "Squares (comprehension): " ++ show squaresList
